### ISSUE TITLE
Unable to use serverless-plugin-datadog when the project uses custom plugins

### ISSUE THREAD

**Issue by afaquejam** [DataDog/serverless-plugin-datadog#43]
## Expected Behavior
The deployment of a serverless service is successful when command `sls deploy` is run.

## Actual Behavior
 ```
Type Error ---------------------------------------------

TypeError: plugins.find is not a function
```

## Steps to Reproduce the Problem

  1. Create a serverless project using the command: `serverless create --template aws-nodejs --path your-service`
  2. In the serverless.yml file, add:
```
plugins:
  localPath: "./serverless_plugins"
  modules:
    - serverless-plugin-datadog
    - any-custom-plugin
```
  3. Deploy: `sls deploy`

## Specifications

  - Serverless Framework version: 1.64.0
  - Datadog Serverless Plugin version: 0.23.0
  - Lambda function runtime (Python 3.7, Node 10, etc.): Node 10
  
## Stacktrace
  
  ```
  Serverless: Auto instrumenting functions with Datadog

  Type Error ---------------------------------------------

  TypeError: plugins.find is not a function
      at Object.hasWebpackPlugin (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/src/util.ts:54:18)
      at Object.entries.map (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/src/layer.ts:73:15)
      at Array.map (<anonymous>)
      at Object.findHandlers (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/src/layer.ts:56:6)
      at ServerlessPlugin.<anonymous> (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/src/index.ts:69:22)
      at Generator.next (<anonymous>)
      at /Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/dist/index.js:15:71
      at new Promise (<anonymous>)
      at __awaiter (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/dist/index.js:11:12)
      at ServerlessPlugin.beforeDeployFunction (/Users/userone/development/scratchpad/js/node_modules/serverless-plugin-datadog/dist/index.js:66:16)
      at BbPromise.reduce (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:490:55)
      at tryCatcher (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
      at Object.gotValue (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/reduce.js:168:18)
      at Object.gotAccum (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/reduce.js:155:25)
      at Object.tryCatcher (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:547:31)
      at Promise._settlePromise (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:604:18)
      at Promise._settlePromise0 (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:649:10)
      at Promise._settlePromises (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:729:18)
      at _drainQueueStep (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:93:12)
      at _drainQueue (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:86:9)
      at Async._drainQueues (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:102:5)
      at Immediate.Async.drainQueues (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:15:14)
      at runCallback (timers.js:705:18)

  ```

**Comment by agocs** [comment]
Hi Aki, thanks for the report. I was able to replicate this on my machine. It looks like no local plugins directory is needed; this is the minimum required to replicate this issue:

 ```
plugins:
  modules:
    - serverless-plugin-datadog
```

It does look like there's some interaction between the Serverless enhanced plugin object and the Datadog plugin. I'll have to research that further and file a bug.

As a workaround, try 

```
plugins:
  - serverless-plugin-datadog
  - ./serverless_plugins/any-custom-plugin
```

This worked on my machine.

Additionally, according to [the Serverless docs](https://www.serverless.com/framework/docs/providers/aws/guide/plugins#service-local-plugin) ,  local plugins can be placed in `.serverless_plugins` and they'll be loaded automatically e.g.

```
plugins:
  - serverless-plugin-datadog
  - any-custom-plugin
```

**Comment by afaquejam** [comment]
Thanks Chris for a quick response. Now the plugin section of our serverless project looks like this:
```
plugins:
  - serverless-plugin-datadog
  - serverless-webpack
  - serverless-offline
  - serverless-pseudo-parameters
  - ../.serverless_plugins/custom-plugin

```

Now I'm getting a different error as show below. I tried installing `datadog-lambda-js` and `dd-trace` but that started giving similar error of `can't resolve ...` for a lot of package. I believe this has got to do withe `serverless-webpack` plugin but I have used `serverless-plugin-datadog` before `serverless-webpack` as suggested in the project's readme file.

```
```

**Comment by kendaop** [comment]
I just want to add that I'm having this issue as well, and I've tried both of the workarounds proposed by @agocs, and I'm still getting the "plugins.find is not a function" error message. I am not using webpack, however....not sure if that makes a difference.

**Comment by zARODz11z** [comment]
Hi @afaquejam and @kendaop A patch for this bug has been released in v2.15.0,  Let us know if it you run into any problems.
